diff --git a/cmake/fix-python-install-scheme.cmake b/cmake/fix-python-install-scheme.cmake
index 4de85a7..960e298 100644
--- a/cmake/fix-python-install-scheme.cmake
+++ b/cmake/fix-python-install-scheme.cmake
@@ -11,7 +11,7 @@
 #
 
 # for IN_LIST to be available
-cmake_minimum_required(VERSION 3.3)
+cmake_minimum_required(VERSION 3.5)
 
 if (DEFINED ENV{DESTDIR})
     set(DESTDIR $ENV{DESTDIR})diff --git a/ext_brotli/CMakeLists.txt b/ext_brotli/CMakeLists.txt
index 9555634..9106951 100644
--- a/ext_brotli/CMakeLists.txt
+++ b/ext_brotli/CMakeLists.txt
@@ -10,7 +10,7 @@ if (MINGW)
     check_cxx_compiler_flag("-mavx512f" CAN_TARGET_AVX512F)
     if (NOT CAN_BUILD_HIGHWAY OR NOT CAN_TARGET_AVX512F)
         message(WARNING "Skipping libjxl, compiler cannot build highway.")
-        return()
+        #return()
     endif()
 endif()
 
diff --git a/ext_highway/CMakeLists.txt b/ext_highway/CMakeLists.txt
index 1c6907a..a068735 100644
--- a/ext_highway/CMakeLists.txt
+++ b/ext_highway/CMakeLists.txt
@@ -10,7 +10,7 @@ if (MINGW)
     check_cxx_compiler_flag("-mavx512f" CAN_TARGET_AVX512F)
     if (NOT CAN_BUILD_HIGHWAY OR NOT CAN_TARGET_AVX512F)
         message(WARNING "Skipping libjxl, compiler cannot build highway.")
-        return()
+        #return()
     endif()
 endif()
 
@@ -21,7 +21,7 @@ ExternalProject_Add( ext_highway
     URL_HASH SHA256=07b3c1ba2c1096878a85a31a5b9b3757427af963b1141ca904db2f9f4afe0bc2
 
     INSTALL_DIR ${EXTPREFIX}
-    CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${EXTPREFIX} DCMAKE_BUILD_TYPE=${GLOBAL_BUILD_TYPE} -DBUILD_TESTING=OFF -DBUILD_STATIC_LIBS=ON -DBUILD_SHARED_LIBS=OFF ${GLOBAL_PROFILE} 
+    CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${EXTPREFIX} DCMAKE_BUILD_TYPE=${GLOBAL_BUILD_TYPE} -DBUILD_TESTING=ON -DBUILD_STATIC_LIBS=ON -DBUILD_SHARED_LIBS=OFF ${GLOBAL_PROFILE} 
 
     UPDATE_COMMAND ""
 )
diff --git a/ext_jpegxl/CMakeLists.txt b/ext_jpegxl/CMakeLists.txt
index a7ad044..a7b0855 100644
--- a/ext_jpegxl/CMakeLists.txt
+++ b/ext_jpegxl/CMakeLists.txt
@@ -10,7 +10,7 @@ if (MINGW)
     check_cxx_compiler_flag("-mavx512f" CAN_TARGET_AVX512F)
     if (NOT CAN_BUILD_HIGHWAY OR NOT CAN_TARGET_AVX512F)
         message(WARNING "Skipping libjxl, compiler cannot build highway.")
-        return()
+        #return()
     endif()
 endif()
 
diff --git a/ext_opus/CMakeLists.txt b/ext_opus/CMakeLists.txt
index ebecdf8..1968fc8 100644
--- a/ext_opus/CMakeLists.txt
+++ b/ext_opus/CMakeLists.txt
@@ -15,6 +15,7 @@ ExternalProject_Add(
         -DCMAKE_BUILD_TYPE=${GLOBAL_BUILD_TYPE}
         -DBUILD_SHARED_LIBS=TRUE
         -DOPUS_STACK_PROTECTOR=OFF
+        -DOPUS_DISABLE_INTRINSICS=TRUE
         ${GLOBAL_PROFILE}
 
     UPDATE_COMMAND ""
diff --git a/ext_pyqt-builder/CMakeLists.txt b/ext_pyqt-builder/CMakeLists.txt
index 27c3769..e4aa0a5 100644
--- a/ext_pyqt-builder/CMakeLists.txt
+++ b/ext_pyqt-builder/CMakeLists.txt
@@ -41,6 +41,8 @@ elseif(WIN32)
         URL_HASH SHA256=3f7a3a2715947a293a97530a76fd59f1309fcb8e57a5830f45c79fe7249b3998
 
         PATCH_COMMAND ${PATCH_COMMAND} -p1 -i ${CMAKE_CURRENT_SOURCE_DIR}/0001-Fix-support-for-MinGW-builds.patch
+            COMMAND git init
+            COMMAND git apply --verbose ${CMAKE_CURRENT_SOURCE_DIR}/0002-bundle-arm64-rt-dlls.patch
 
         CONFIGURE_COMMAND ""
         BUILD_COMMAND ""
diff --git a/ext_python/CMakeLists.txt b/ext_python/CMakeLists.txt
index 8c78c2f..e0bc761 100644
--- a/ext_python/CMakeLists.txt
+++ b/ext_python/CMakeLists.txt
@@ -101,8 +101,8 @@ if (UNIX)
 elseif(WIN32)
     ExternalProject_Add( ext_python
         DOWNLOAD_DIR ${EXTERNALS_DOWNLOAD_DIR}
-        URL https://www.python.org/ftp/python/3.13.5/python-3.13.5-embed-amd64.zip
-        URL_MD5 370a345dbea8bbc1830a2385f24632d2
+        URL https://www.python.org/ftp/python/3.13.14/python-3.13.14-embed-arm64.zip
+        # URL_MD5 370a345dbea8bbc1830a2385f24632d2
 
         CONFIGURE_COMMAND ""
         BUILD_COMMAND ""
diff --git a/ext_sip/0001-Fix-version-tuple-w-none.patch b/ext_sip/0001-Fix-version-tuple-w-none.patch
new file mode 100644
index 0000000..2521456
--- /dev/null
+++ b/ext_sip/0001-Fix-version-tuple-w-none.patch
@@ -0,0 +1,15 @@
+diff --git a/sipbuild/bindings.py b/sipbuild/bindings.py
+index 783d21d..7d100d1 100644
+--- a/sipbuild/bindings.py
++++ b/sipbuild/bindings.py
+@@ -151,7 +151,7 @@ spec, modules, sip_files = parse(self.sip_file, SIP_VERSION, encoding,
+                 project.sip_module)
+ 
+         # Update the target ABI for the project.
+-        if project.target_abi is None or project.target_abi < spec.target_abi:
++        if (project.target_abi is None) or (None in project.target_abi) or project.target_abi < spec.target_abi:
+             project.target_abi = spec.target_abi
+ 
+         # Resolve the types.
+--
+
diff --git a/ext_sip/CMakeLists.txt b/ext_sip/CMakeLists.txt
index fafc5a4..e52a536 100644
--- a/ext_sip/CMakeLists.txt
+++ b/ext_sip/CMakeLists.txt
@@ -45,6 +45,7 @@ elseif (WIN32)
         URL_HASH SHA256=fa0515697d4c98dbe04d9e898d816de1427e5b9ae5d0e152169109fd21f5d29c
 
         PATCH_COMMAND ${PATCH_COMMAND} -p1 -i ${CMAKE_CURRENT_SOURCE_DIR}/0001-Fix-distinfo.py-to-support-prefix-option-on-Windows-.patch
+            COMMAND ${PATCH_COMMAND} -p1 -i ${CMAKE_CURRENT_SOURCE_DIR}/0001-Fix-version-tuple-w-none.patch
 
         CONFIGURE_COMMAND ""
         BUILD_COMMAND ""
diff --git a/ext_vpx/0000-windows-python-workaround.patch b/ext_vpx/0000-windows-python-workaround.patch
new file mode 100644
index 0000000..6c70c17
--- /dev/null
+++ b/ext_vpx/0000-windows-python-workaround.patch
@@ -0,0 +1,52 @@
+diff --git a/meson.build b/meson.build
+index 3a023f00b..1f1e53c86 100644
+--- a/meson.build
++++ b/meson.build
+@@ -1672,14 +1672,14 @@ if asm_conversion_exe.found()
+ 	stdinout_wrapper_exe = find_program('meson/stdinout_wrapper.py', required: true)
+ 	asm_transformer = generator(
+ 		stdinout_wrapper_exe,
+-		arguments: ['--input', '@INPUT@', '--output', '@OUTPUT@', asm_conversion_exe.full_path()] + asm_conversion_cmd,
++		arguments: ['--input', '@INPUT@', '--output', '@OUTPUT@', 'python', asm_conversion_exe.full_path()] + asm_conversion_cmd,
+ 		output: '@0@@1@'.format('@BASENAME@', as_sfx)
+ 	)
+ endif
+ 
+ if features.get('x86', false) or features.get('x86_64', false)
+ 	config_asm = configure_file(
+-		command: [config_asm_generator, '@INPUT@', '@OUTPUT@', '--format=yasm'],
++		command: ['python', config_asm_generator, '@INPUT@', '@OUTPUT@', '--format=yasm'],
+ 		input: vpx_config_h,
+ 		output: 'vpx_config.asm',
+ 	)
+@@ -1696,19 +1696,19 @@ if features.get('x86', false) or features.get('x86_64', false)
+ 	)
+ elif asm_conversion_exe.found()
+ 	ads_config_asm = configure_file(
+-		command: [config_asm_generator, '@INPUT@', '@OUTPUT@'],
++		command: ['python', config_asm_generator, '@INPUT@', '@OUTPUT@'],
+ 		input: vpx_config_h,
+ 		output: 'vpx_config.adstmp',
+ 	)
+ 	config_asm = configure_file(
+-		command: [stdinout_wrapper_exe, '--input', '@INPUT@', '--output', '@OUTPUT@', asm_conversion_exe.full_path()] + asm_conversion_cmd,
++		command: [stdinout_wrapper_exe, '--input', '@INPUT@', '--output', '@OUTPUT@', 'python', asm_conversion_exe.full_path()] + asm_conversion_cmd,
+ 		input: ads_config_asm,
+ 		output: 'vpx_config.asm',
+ 	)
+ 	asm_compiler = disabler()
+ else
+ 	config_asm = configure_file(
+-		command: [config_asm_generator, '@INPUT@', '@OUTPUT@'],
++		command: ['python', config_asm_generator, '@INPUT@', '@OUTPUT@'],
+ 		input: vpx_config_h,
+ 		output: 'vpx_config.asm',
+ 	)
+@@ -1781,6 +1781,7 @@ endforeach
+ gen_def_exe = find_program('meson/gen_def.py', required: true)
+ 
+ gen_def_cmd = [
++	'python',
+ 	gen_def_exe,
+ 	'--name=vpx',
+ 	'--out=@OUTPUT@'
\ No newline at end of file
diff --git a/ext_vpx/CMakeLists.txt b/ext_vpx/CMakeLists.txt
index 47764de..3011a4f 100644
--- a/ext_vpx/CMakeLists.txt
+++ b/ext_vpx/CMakeLists.txt
@@ -340,6 +40,7 @@ kis_ExternalProject_Add_with_separate_builds_apple(
     # I guess it is related to the MSVC support. Since we use clang for
     # our builds, it should be fine just to ignore this check
     PATCH_COMMAND ${PATCH_COMMAND} -p1 -i ${CMAKE_CURRENT_SOURCE_DIR}/0001-Allow-shared-libraries-everywhere.patch
+          COMMAND ${PATCH_COMMAND} -p1 -i ${CMAKE_CURRENT_SOURCE_DIR}/0000-windows-python-workaround.patch
 
     CONFIGURE_ARGS --prefix=${EXTPREFIX}
             -Ddefault_library=shared
diff --git a/latest/krita-deps.yml b/latest/krita-deps.yml
index e4cce67..4b18e2c 100644
--- a/latest/krita-deps.yml
+++ b/latest/krita-deps.yml
@@ -3,7 +3,7 @@
     "base/boost": "master"
     "base/lcms2": "master"
     "base/png": "master"
-    "base/nasm": "master"
+    # "base/nasm": "master"
     "base/eigen3": "master"
     "base/expat": "master"
     "base/exiv2": "master"
@@ -139,11 +139,11 @@
 - 'on': ['Windows']
   'require':
     "base/patch": "master"
-    "base/strawberryperl": "master"
+    # "base/strawberryperl": "master"
     "base/zlib": "master"
     "base/googleangle": "master"
     "base/drmingw": "master"
-    "base/icoutils": "master"
+    # "base/icoutils": "master"
     "base/dlfcn_win32": "master"
     "base/gperf": "master"
 
@@ -155,7 +155,7 @@
     # TODO: check if we actually need that on macOS (required by ki18n)
     "base/gettext": "master"
     # TODO: check if we actually need that on macOS (required by Qt and python)
-    "base/openssl": "master"
+    #"base/openssl": "master"
 
 - 'on': ['Linux', 'MacOS']
   'require':
diff --git a/requirements.txt b/requirements.txt
index 475ea7e..dafb0dc 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -432,3 +432,5 @@ setuptools==82.0.1 \
     --hash=sha256:7d872682c5d01cfde07da7bccc7b65469d3dca203318515ada1de5eda35efbf9 \
     --hash=sha256:a59e362652f08dcd477c78bb6e7bd9d80a7995bc73ce773050228a348ce2e5bb
     # via -r requirements.txt.in
+
+pefile==2024.8.26
diff --git a/requirements.txt.in b/requirements.txt.in
index d39062d..02fa60d 100644
--- a/requirements.txt.in
+++ b/requirements.txt.in
@@ -8,3 +8,4 @@ ply
 tomli
 wheel
 meson
+pefile
