diff --git a/ext_boost/CMakeLists.txt b/ext_boost/CMakeLists.txt
index 10ba997..4f01c03 100644
--- a/ext_boost/CMakeLists.txt
+++ b/ext_boost/CMakeLists.txt
@@ -13,20 +13,20 @@ if(WIN32)
         string(REGEX REPLACE "([0-9])\\.([0-9])(\\.[0-9])?" "\\1"
             KRITA_boost_COMPILER_VERSION ${CMAKE_CXX_COMPILER_VERSION})
 
-        set (BUILD_ARGS -j${SUBMAKE_JOBS} linkflags="${SECURITY_SHARED_LINKER_FLAGS}" --with-system --build-dir=build-dir toolset=gcc variant=release link=shared threading=multi architecture=x86 address-model=64)
+        set (BUILD_ARGS -j${SUBMAKE_JOBS} linkflags="${SECURITY_SHARED_LINKER_FLAGS}" --with-system --with-test --debug --build-dir=build-dir toolset=gcc variant=release link=shared threading=multi architecture=arm address-model=64)
 
         ExternalProject_Add(
             ext_boost
 
             DOWNLOAD_DIR ${EXTERNALS_DOWNLOAD_DIR}
 
-            URL https://archives.boost.io/release/1.80.0/source/boost_1_80_0.tar.bz2
-            URL_HASH SHA256=1e19565d82e43bc59209a168f5ac899d3ba471d55c7610c677d4ccf2c9c500c0
+            URL https://archives.boost.io/release/1.87.0/source/boost_1_87_0.tar.bz2
+            # URL_HASH SHA256=1e19565d82e43bc59209a168f5ac899d3ba471d55c7610c677d4ccf2c9c500c0
 
-            PATCH_COMMAND ${PATCH_COMMAND} -p1 -i ${CMAKE_CURRENT_SOURCE_DIR}/0001-Fix-Krita-builds-with-Clang.patch
-                  COMMAND ${PATCH_COMMAND} -p1 -i ${CMAKE_CURRENT_SOURCE_DIR}/0001-Fix-building-Boost.Build-with-a-chosen-toolset.patch
-                  COMMAND ${PATCH_COMMAND} -p1 -i ${CMAKE_CURRENT_SOURCE_DIR}/0001-Fix-compilation-with-clang-15.patch
-                  COMMAND ${PATCH_COMMAND} -p1 -i ${CMAKE_CURRENT_SOURCE_DIR}/0001-mpl-Use-static-const-for-next-prior-from-C-11.patch
+            # PATCH_COMMAND ${PATCH_COMMAND} -p1 -i ${CMAKE_CURRENT_SOURCE_DIR}/0001-Fix-Krita-builds-with-Clang.patch
+            #       COMMAND ${PATCH_COMMAND} -p1 -i ${CMAKE_CURRENT_SOURCE_DIR}/0001-Fix-building-Boost.Build-with-a-chosen-toolset.patch
+            #       COMMAND ${PATCH_COMMAND} -p1 -i ${CMAKE_CURRENT_SOURCE_DIR}/0001-Fix-compilation-with-clang-15.patch
+            #       COMMAND ${PATCH_COMMAND} -p1 -i ${CMAKE_CURRENT_SOURCE_DIR}/0001-mpl-Use-static-const-for-next-prior-from-C-11.patch
 
             CONFIGURE_COMMAND <SOURCE_DIR>/bootstrap.bat gcc --prefix=${EXTPREFIX}
             BUILD_COMMAND ${CMAKE_COMMAND} -DPREFIX=${EXTPREFIX} -P ${CMAKE_CURRENT_SOURCE_DIR}/run-b2-with-destdir.cmake -- ${BUILD_ARGS}
@@ -38,7 +38,7 @@ if(WIN32)
         ExternalProject_Add_Step(
             ext_boost
             pre_install
-            COMMAND ${CMAKE_COMMAND} -DSRC=<SOURCE_DIR>/build-dir/boost/bin.v2/libs/system/build/gcc-${KRITA_boost_COMPILER_VERSION}/release/threading-multi/visibility-hidden/libboost_system-mgw${KRITA_boost_COMPILER_VERSION}-mt-x64-1_80.dll -DDST=${EXTPREFIX}/bin -P ${KRITA_CI_INSTALL}
+            COMMAND ${CMAKE_COMMAND} -DSRC=<SOURCE_DIR>/build-dir/boost/bin.v2/libs/system/build/gcc-${KRITA_boost_COMPILER_VERSION}/release/arm_64/threadapi-win32/threading-multi/visibility-hidden/libboost_system-mgw${KRITA_boost_COMPILER_VERSION}-mt-a64-1_87.dll -DDST=${EXTPREFIX}/bin -P ${KRITA_CI_INSTALL}
             DEPENDERS install
         )
         if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
@@ -73,8 +73,8 @@ if(WIN32)
             ext_boost
             DOWNLOAD_DIR ${EXTERNALS_DOWNLOAD_DIR}
 
-            URL https://archives.boost.io/release/1.80.0/source/boost_1_80_0.tar.bz2
-            URL_HASH SHA256=1e19565d82e43bc59209a168f5ac899d3ba471d55c7610c677d4ccf2c9c500c0
+            URL https://archives.boost.io/release/1.87.0/source/boost_1_87_0.tar.bz2
+            # URL_HASH SHA256=1e19565d82e43bc59209a168f5ac899d3ba471d55c7610c677d4ccf2c9c500c0
 
             CONFIGURE_COMMAND <SOURCE_DIR>/bootstrap.bat msvc --prefix=${EXTPREFIX}
             BUILD_COMMAND ${CMAKE_COMMAND} -DPREFIX=${EXTPREFIX} -P ${CMAKE_CURRENT_SOURCE_DIR}/run-b2-with-destdir.cmake -- ${BUILD_ARGS}
@@ -134,8 +134,8 @@ else()
     endif()
     ExternalProject_Add( ext_boost
         DOWNLOAD_DIR ${EXTERNALS_DOWNLOAD_DIR}
-        URL https://archives.boost.io/release/1.80.0/source/boost_1_80_0.tar.bz2
-        URL_HASH SHA256=1e19565d82e43bc59209a168f5ac899d3ba471d55c7610c677d4ccf2c9c500c0
+        URL https://archives.boost.io/release/1.87.0/source/boost_1_87_0.tar.bz2
+        # URL_HASH SHA256=1e19565d82e43bc59209a168f5ac899d3ba471d55c7610c677d4ccf2c9c500c0
 
         PATCH_COMMAND ${PATCH_COMMAND} -p1 -i ${CMAKE_CURRENT_SOURCE_DIR}/0001-Fix-compilation-with-clang-15.patch
               COMMAND ${PATCH_COMMAND} -p1 -i ${CMAKE_CURRENT_SOURCE_DIR}/0001-mpl-Use-static-const-for-next-prior-from-C-11.patch
diff --git a/ext_brotli/CMakeLists.txt b/ext_brotli/CMakeLists.txt
index 64dbe95..7f7af60 100644
--- a/ext_brotli/CMakeLists.txt
+++ b/ext_brotli/CMakeLists.txt
@@ -13,7 +13,7 @@ if (MINGW)
     check_cxx_compiler_flag("-mavx512f" CAN_TARGET_AVX512F)
     if (NOT CAN_BUILD_HIGHWAY OR NOT CAN_TARGET_AVX512F)
         message(WARNING "Skipping libjxl, compiler cannot build highway.")
-        return()
+        #return()
     endif()
 endif()
 
diff --git a/ext_freetype/CMakeLists.txt b/ext_freetype/CMakeLists.txt
index e0504de..50857c1 100644
--- a/ext_freetype/CMakeLists.txt
+++ b/ext_freetype/CMakeLists.txt
@@ -58,7 +58,7 @@ kis_ExternalProject_Add_with_separate_builds_apple(
     CONFIGURE_ARGS --prefix=${BOOTSTRAP_PREFIX}
         -Dglib=disabled
         -Dgobject=disabled
-        -Dtests=disabled
+        -Dtests=enabled
         -Dfreetype=enabled
         -Dpkgconfig.relocatable=true
         --pkg-config-path ${_mesonified_pkg_config_path}
@@ -84,6 +84,7 @@ ExternalProject_Add( ext_freetype
         -DFT_REQUIRE_HARFBUZZ=ON
         -DFT_REQUIRE_PNG=ON
         -DBUILD_SHARED_LIBS=TRUE
+        -DBUILD_TESTING=ON
         -DCMAKE_PREFIX_PATH=${EXTPREFIX}
         ${GLOBAL_PROFILE}
 
diff --git a/ext_poppler/0000-clang-basic_string.patch b/ext_poppler/0000-clang-basic_string.patch
new file mode 100644
index 0000000..dca88a2
--- /dev/null
+++ b/ext_poppler/0000-clang-basic_string.patch
@@ -0,0 +1,13 @@
+diff --git a/cpp/poppler-global.h b/cpp/poppler-global.h
+index 7d3c890..26a2d00 100644
+--- a/cpp/poppler-global.h
++++ b/cpp/poppler-global.h
+@@ -96,7 +96,7 @@ typedef unsigned int /* time_t */ time_type;
+ #    pragma warning(push)
+ #    pragma warning(disable : 4251) /* class 'A' needs to have dll interface for to be used by clients of class 'B'. */
+ #endif
+-class POPPLER_CPP_EXPORT ustring : public std::basic_string<unsigned short>
++class POPPLER_CPP_EXPORT ustring : public std::basic_string<char16_t>
+ {
+ public:
+     ustring();
\ No newline at end of file
diff --git a/ext_poppler/CMakeLists.txt b/ext_poppler/CMakeLists.txt
index 22d6d84..ad67b64 100644
--- a/ext_poppler/CMakeLists.txt
+++ b/ext_poppler/CMakeLists.txt
@@ -13,6 +13,7 @@ ExternalProject_Add( ext_poppler
 
     PATCH_COMMAND ${PATCH_COMMAND} -p1 -i ${CMAKE_CURRENT_SOURCE_DIR}/0001-fix_sed_invocation.patch
             COMMAND ${PATCH_COMMAND} -p1 -i ${CMAKE_CURRENT_SOURCE_DIR}/0001-macos-add-missing-system-header.patch
+            COMMAND ${PATCH_COMMAND} -p1 -i ${CMAKE_CURRENT_SOURCE_DIR}/0000-clang-basic_string.patch
 
 	INSTALL_DIR ${EXTPREFIX}
 	CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${EXTPREFIX} -DCMAKE_BUILD_TYPE=${GLOBAL_BUILD_TYPE} ${GLOBAL_PROFILE} -DBUILD_GTK_TESTS=OFF -DBUILD_QT5_TESTS=FALSE -DBUILD_CPP_TESTS=FALSE  -DENABLE_UTILS=FALSE -DENABLE_GLIB=FALSE -DENABLE_LIBOPENJPEG=none -DWITH_NSS3=OFF -DBUILD_MANUAL_TESTS=OFF
diff --git a/ext_pyqt-builder/CMakeLists.txt b/ext_pyqt-builder/CMakeLists.txt
index 27c3769..e4aa0a5 100644
--- a/ext_pyqt-builder/CMakeLists.txt
+++ b/ext_pyqt-builder/CMakeLists.txt
@@ -44,6 +44,8 @@ elseif(WIN32)
         URL_HASH SHA256=3f7a3a2715947a293a97530a76fd59f1309fcb8e57a5830f45c79fe7249b3998
 
         PATCH_COMMAND ${PATCH_COMMAND} -p1 -i ${CMAKE_CURRENT_SOURCE_DIR}/0001-Fix-support-for-MinGW-builds.patch
+            COMMAND git init
+            COMMAND git apply --verbose ${CMAKE_CURRENT_SOURCE_DIR}/0002-bundle-arm64-rt-dlls.patch
 
         CONFIGURE_COMMAND ""
         BUILD_COMMAND ""
diff --git a/ext_python/CMakeLists.txt b/ext_python/CMakeLists.txt
index 8c78c2f..e0bc761 100644
--- a/ext_python/CMakeLists.txt
+++ b/ext_python/CMakeLists.txt
@@ -104,15 +104,15 @@ if (UNIX)
 elseif(WIN32)
     ExternalProject_Add( ext_python
         DOWNLOAD_DIR ${EXTERNALS_DOWNLOAD_DIR}
-        URL https://www.python.org/ftp/python/3.13.5/python-3.13.5-embed-amd64.zip
-        URL_MD5 370a345dbea8bbc1830a2385f24632d2
+        URL "<python-3.10-embed-arm64.zip here>"
+        # URL_MD5 370a345dbea8bbc1830a2385f24632d2
 
         CONFIGURE_COMMAND ""
         BUILD_COMMAND ""
         INSTALL_COMMAND ${CMAKE_COMMAND} -E echo deploying python3 64-bit binary
                 COMMAND ${CMAKE_COMMAND} -DSRC=<SOURCE_DIR>/ -DDST=${EXTPREFIX}/python -P ${KRITA_CI_INSTALL_DIRECTORY}
                 COMMAND ${CMAKE_COMMAND} -DSRC=<SOURCE_DIR>/python3.dll -DDST=${EXTPREFIX}/bin -P  ${KRITA_CI_INSTALL}
-                COMMAND ${CMAKE_COMMAND} -DSRC=<SOURCE_DIR>/python313.dll -DDST=${EXTPREFIX}/bin -P  ${KRITA_CI_INSTALL}
+                COMMAND ${CMAKE_COMMAND} -DSRC=<SOURCE_DIR>/python310.dll -DDST=${EXTPREFIX}/bin -P  ${KRITA_CI_INSTALL}
                 COMMAND ${CMAKE_COMMAND} -DSRC=<SOURCE_DIR>/vcruntime140.dll -DDST=${EXTPREFIX}/bin -P  ${KRITA_CI_INSTALL}
                 COMMAND ${CMAKE_COMMAND} -DSRC=<SOURCE_DIR>/vcruntime140_1.dll -DDST=${EXTPREFIX}/bin -P  ${KRITA_CI_INSTALL}
                 COMMAND ${CMAKE_COMMAND} -DSRC=${CMAKE_CURRENT_SOURCE_DIR}/sitecustomize.py -DDST=${EXTPREFIX}/lib/site-packages/ -P  ${KRITA_CI_INSTALL}
diff --git a/ext_sip/0001-Fix-version-tuple-w-none.patch b/ext_sip/0001-Fix-version-tuple-w-none.patch
new file mode 100644
index 0000000..2521456
--- /dev/null
+++ b/ext_sip/0001-Fix-version-tuple-w-none.patch
@@ -0,0 +1,15 @@
+diff --git a/sipbuild/bindings.py b/sipbuild/bindings.py
+index 783d21d..7d100d1 100644
+--- a/sipbuild/bindings.py
++++ b/sipbuild/bindings.py
+@@ -151,7 +151,7 @@ spec, modules, sip_files = parse(self.sip_file, SIP_VERSION, encoding,
+                 project.sip_module)
+ 
+         # Update the target ABI for the project.
+-        if project.target_abi is None or project.target_abi < spec.target_abi:
++        if (project.target_abi is None) or (None in project.target_abi) or project.target_abi < spec.target_abi:
+             project.target_abi = spec.target_abi
+ 
+         # Resolve the types.
+--
+
diff --git a/ext_sip/CMakeLists.txt b/ext_sip/CMakeLists.txt
index fafc5a4..e52a536 100644
--- a/ext_sip/CMakeLists.txt
+++ b/ext_sip/CMakeLists.txt
@@ -48,6 +48,7 @@ elseif (WIN32)
         URL_HASH SHA256=fa0515697d4c98dbe04d9e898d816de1427e5b9ae5d0e152169109fd21f5d29c
 
         PATCH_COMMAND ${PATCH_COMMAND} -p1 -i ${CMAKE_CURRENT_SOURCE_DIR}/0001-Fix-distinfo.py-to-support-prefix-option-on-Windows-.patch
+            COMMAND ${PATCH_COMMAND} -p1 -i ${CMAKE_CURRENT_SOURCE_DIR}/0001-Fix-version-tuple-w-none.patch
 
         CONFIGURE_COMMAND ""
         BUILD_COMMAND ""
diff --git a/ext_vpx/0000-windows-python-workaround.patch b/ext_vpx/0000-windows-python-workaround.patch
new file mode 100644
index 0000000..6c70c17
--- /dev/null
+++ b/ext_vpx/0000-windows-python-workaround.patch
@@ -0,0 +1,52 @@
+diff --git a/meson.build b/meson.build
+index 3a023f00b..1f1e53c86 100644
+--- a/meson.build
++++ b/meson.build
+@@ -1672,14 +1672,14 @@ if asm_conversion_exe.found()
+ 	stdinout_wrapper_exe = find_program('meson/stdinout_wrapper.py', required: true)
+ 	asm_transformer = generator(
+ 		stdinout_wrapper_exe,
+-		arguments: ['--input', '@INPUT@', '--output', '@OUTPUT@', asm_conversion_exe.full_path()] + asm_conversion_cmd,
++		arguments: ['--input', '@INPUT@', '--output', '@OUTPUT@', 'python', asm_conversion_exe.full_path()] + asm_conversion_cmd,
+ 		output: '@0@@1@'.format('@BASENAME@', as_sfx)
+ 	)
+ endif
+ 
+ if features.get('x86', false) or features.get('x86_64', false)
+ 	config_asm = configure_file(
+-		command: [config_asm_generator, '@INPUT@', '@OUTPUT@', '--format=yasm'],
++		command: ['python', config_asm_generator, '@INPUT@', '@OUTPUT@', '--format=yasm'],
+ 		input: vpx_config_h,
+ 		output: 'vpx_config.asm',
+ 	)
+@@ -1696,19 +1696,19 @@ if features.get('x86', false) or features.get('x86_64', false)
+ 	)
+ elif asm_conversion_exe.found()
+ 	ads_config_asm = configure_file(
+-		command: [config_asm_generator, '@INPUT@', '@OUTPUT@'],
++		command: ['python', config_asm_generator, '@INPUT@', '@OUTPUT@'],
+ 		input: vpx_config_h,
+ 		output: 'vpx_config.adstmp',
+ 	)
+ 	config_asm = configure_file(
+-		command: [stdinout_wrapper_exe, '--input', '@INPUT@', '--output', '@OUTPUT@', asm_conversion_exe.full_path()] + asm_conversion_cmd,
++		command: [stdinout_wrapper_exe, '--input', '@INPUT@', '--output', '@OUTPUT@', 'python', asm_conversion_exe.full_path()] + asm_conversion_cmd,
+ 		input: ads_config_asm,
+ 		output: 'vpx_config.asm',
+ 	)
+ 	asm_compiler = disabler()
+ else
+ 	config_asm = configure_file(
+-		command: [config_asm_generator, '@INPUT@', '@OUTPUT@'],
++		command: ['python', config_asm_generator, '@INPUT@', '@OUTPUT@'],
+ 		input: vpx_config_h,
+ 		output: 'vpx_config.asm',
+ 	)
+@@ -1781,6 +1781,7 @@ endforeach
+ gen_def_exe = find_program('meson/gen_def.py', required: true)
+ 
+ gen_def_cmd = [
++	'python',
+ 	gen_def_exe,
+ 	'--name=vpx',
+ 	'--out=@OUTPUT@'
\ No newline at end of file
diff --git a/ext_vpx/CMakeLists.txt b/ext_vpx/CMakeLists.txt
index 47764de..3011a4f 100644
--- a/ext_vpx/CMakeLists.txt
+++ b/ext_vpx/CMakeLists.txt
@@ -32,6 +32,7 @@ kis_ExternalProject_Add_with_separate_builds_apple(
     # I guess it is related to the MSVC support. Since we use clang for
     # our builds, it should be fine just to ignore this check
     PATCH_COMMAND ${PATCH_COMMAND} -p1 -i ${CMAKE_CURRENT_SOURCE_DIR}/0001-Allow-shared-libraries-everywhere.patch
+          COMMAND ${PATCH_COMMAND} -p1 -i ${CMAKE_CURRENT_SOURCE_DIR}/0000-windows-python-workaround.patch
 
     CONFIGURE_ARGS --prefix=${EXTPREFIX}
             -Ddefault_library=shared
diff --git a/latest/krita-deps.yml b/latest/krita-deps.yml
index e4cce67..4b18e2c 100644
--- a/latest/krita-deps.yml
+++ b/latest/krita-deps.yml
@@ -3,7 +3,7 @@
     "base/boost": "master"
     "base/lcms2": "master"
     "base/png": "master"
-    "base/nasm": "master"
+    # "base/nasm": "master"
     "base/eigen3": "master"
     "base/expat": "master"
     "base/exiv2": "master"
@@ -30,9 +30,9 @@
     "base/zug": "master"
     "base/immer": "master"
     "base/lager": "master"
-    "base/highway": "master"
+    # "base/highway": "master"
     "base/brotli": "master"
-    "base/jpegxl": "master"
+    #"base/jpegxl": "master"
     "base/extra_cmake_modules": "master"
     "base/kconfig": "master"
     "base/karchive": "master"
@@ -138,11 +138,11 @@
 - 'on': ['Windows']
   'require':
     "base/patch": "master"
-    "base/strawberryperl": "master"
+    # "base/strawberryperl": "master"
     "base/zlib": "master"
     "base/googleangle": "master"
     "base/drmingw": "master"
-    "base/icoutils": "master"
+    # "base/icoutils": "master"
     "base/dlfcn_win32": "master"
 
 - 'on': ['Windows', 'MacOS']
@@ -153,7 +153,7 @@
     # TODO: check if we actually need that on macOS (required by ki18n)
     "base/gettext": "master"
     # TODO: check if we actually need that on macOS (required by Qt and python)
-    "base/openssl": "master"
+    #"base/openssl": "master"
 
 - 'on': ['Linux', 'MacOS']
   'require':
